redhat-developer/try-in-dev-spaces-browser-extension
# Update options page UI

Fixes https://github.com/redhat-developer/try-in-dev-spaces-browser-extension/issues/23, https://github.com/redhat-developer/try-in-dev-spaces-browser-extension/issues/24, https://github.com/redhat-developer/try-in-dev-spaces-browser-extension/issues/18

<details>
<summary>Demo on Google Chrome:</summary>

https://user-images.githubusercontent.com/83611742/222549889-cd0b8a54-5f08-4010-8e03-3e72cf5a7477.mov

</details>

<details>
<summary>Edge</summary>

<img width="455" alt="image" src="https://user-images.githubusercontent.com/83611742/222545394-ec3a95f0-2a96-4771-b87d-6d8247ab88da.png">

</details>

<details>
<summary>Firefox:</summary>

<img width="727" alt="Screenshot 2023-03-02 at 11 12 55 AM" src="https://user-images.githubusercontent.com/83611742/222544986-d275cb8b-ad3a-4b35-a92d-bb037fb84cdd.png">

</details>

<details>
<summary>Safari:</summary>

https://user-images.githubusercontent.com/83611742/222545146-61dd6a44-11b4-4d72-9f8e-c6a70eff3e73.mov

</details>


To test this PR:

1. Checkout this PR and run `yarn && yarn build`
2. Sideload the extension located in `dist/chromium` into Google Chrome
3. Verify that the options page features are working as before (Add new devspaces hostname, remove hostname, add hostname again, set default, confirm that changes are working by checking the injected button on a GitHub repo page)
4. Verify that the malformed URLs cannot be added
5. Run `yarn test` to run all tests
